libs/remote_storage/src/s3_bucket.rs

```diff
@@ -83,10 +87,23 @@ impl S3Bucket {
             .or_else("imds", ImdsCredentialsProvider::builder().build())
         };
 
+        // AWS SDK requires us to specify how the RetryConfig should sleep when it wants to back off
+        let sleep_impl: Arc<dyn AsyncSleep> = Arc::new(TokioSleep::new());
+
+        // We do our own retries (see [`backoff::retry`]).  However, for the AWS SDK to enable rate limiting in response to throttling
+        // responses (e.g. 429 on too many ListObjectsv2 requests), we must provide a retry config.  We set it to use at most one
+        // attempt, and enable 'Adaptive' mode, which causes rate limiting to be enabled.
+        let mut retry_config = RetryConfigBuilder::new();
+        retry_config
+            .set_max_attempts(Some(1))
+            .set_mode(Some(RetryMode::Adaptive));
```

Comment:
Can we see somewhere that this will actually do the shared sleep *after* the initial and only connection fails?

Reply:
It is sleeping before making retry.

Reply:
@koivunej so the first call will go through get 429, but won't retry as max attempts will be 1
Since we have backoff::retry, a new call will be created. but this time it will sleep before calling out to aws since it was configured with adaptive?